server/src/main/java/org/opensearch/search/backpressure/SearchBackpressureState.java

```diff
@@ -54,4 +71,28 @@ public long getLimitReachedCount() {
     long incrementLimitReachedCount() {
         return limitReachedCount.incrementAndGet();
     }
+
+    public AtomicReference<TokenBucket> getRateLimiter() {
+        return rateLimiter;
+    }
+
+    public AtomicReference<TokenBucket> getRatioLimiter() {
+        return ratioLimiter;
+    }
+
+    void onCancellationBurstChanged(double cancellationBurst) {
+        this.cancellationBurst = cancellationBurst;
+        onCancellationRateChanged(cancellationRate);
+        onCancellationRatioChanged(cancellationRatio);
+    }
+
+    void onCancellationRateChanged(double cancellationRate) {
+        this.cancellationRate = cancellationRate;
+        rateLimiter.set(new TokenBucket(timeNanosSupplier, cancellationRate, cancellationBurst));
+    }
+
+    void onCancellationRatioChanged(double cancellationRatio) {
+        this.cancellationRatio = cancellationRatio;
+        ratioLimiter.set(new TokenBucket(this::getCompletionCount, cancellationRatio, cancellationBurst));
+    }
```

Comment:
not sure if you need concurrency handling here but for my understanding in case you feel you do, what are your thoughts on compare and set as against set

Reply:
@Bukhtawar can you please explain why `compareAndSet` is required here? Is it because we want to avoid re-initialisation of limiter if the same cancellationRatio value is set again or something else?

Reply:
As per my discussion with @ketanv3 , no need of `compareAndSet` here.